lib/user_space/user_x86_64_atomic.h

```diff
@@ -167,11 +254,43 @@ static inline bool m0_atomic64_cas(int64_t * loc, int64_t oldval, int64_t newval
 		     : "r" (newval), "0" (oldval)
 		     : "memory");
 	return val == oldval;
+#elif defined CONFIG_AARCH64 /*aarch64*/
+/*	unsigned long	tmp;
+	int64_t	 old=0;
+
+	M0_CASSERT(8 == sizeof oldval);
+
+	asm volatile("// atomic64_cas_return \n"		\
+		     "  prfm    pstl1strm, %[v]\n"		\
+		     "1:ldxr    %[old], %[v]\n"  		\
+		     "  eor     %[tmp], %[oldval], %[old]\n"	\
+		     "  cbnz    %[tmp], 2f\n"			\
```

Comment:
Actually this  part of code being  processor  specific assembly instruction , This would  act more comprehensible to other while porting to other  platform. Since this code has few flaws, not debugged yet. it is not working as of now.   The  gcc specific  routine is called in place of it   which has no  public source code available. So it is kept as part of documentation for future reference and implementation .

Reply:
Please, add a comment about this then.

Reply:
added